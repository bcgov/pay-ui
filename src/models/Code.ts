// To handle codes tables such as SuspensionReasonCode, etc
export interface Code {
    code?: string;
    description?: string;
}

export interface PaymentMethod {
    code: string;
    description: string;
    partialRefund?: boolean;
    refundableStatuses?: string;
}
