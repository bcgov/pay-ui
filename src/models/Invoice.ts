// Model for storing invoice of a routing slip
export interface LineItem {
  description?: string
  filingFees?: number
  futureEffectiveFees?: number
  gst?: number
  id?: number
  priorityFees?: number
  pst?: number
  quantity?: number
  serviceFees?: number
  statusCode?: string
  total?: number
  waivedBy?: string
  waivedFees?: number
}

export interface Reference {
  id?: number
  invoiceNumber?: string
  statusCode?: string
}

export interface Detail {
  label?: string
  value?: string
}

export interface Invoice {
  businessIdentifier?: string
  corpTypeCode?: string
  createdBy?: string
  createdOn?: string
  createdName?: string
  id?: number
  lineItems?: LineItem[]
  detail?: Detail[]
  paid?: number,
  paymentAccount?: {
    accountId?: string
    accountName?: string
    billable?: boolean
  }
  folioNumber?: string
  paymentMethod?: string
  references?: Reference[]
  refund?: number
  routingSlip?: string
  serviceFees?: string
  statusCode?: string
  total?: number,
  details?: Detail[]
}

export interface InvoiceList {
  consInvNumber?: string
  invoiceNumber: string
  invoices: Invoice[],
  paymentMethod: string
  paymentSystem: string
  statusCode: string
  invoiceAmount: number
  paidAmount: number
}

export interface InvoiceListResponse {
  items: InvoiceList[]
  limit: number
  page: number
  total: number
}

// For displaying transaction data table
export interface InvoiceDisplay {
  createdOn?: string
  invoiceNumber?: string
  statusCode?: string
  total?: number
  createdName?: string
  createdBy?: string
  description?: string[]
  id?: number
}
