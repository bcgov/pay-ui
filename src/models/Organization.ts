import { Contact } from './contact'

export interface Organization {
  id?: number
}

export interface Organizations{
  orgs : Organization[]
}

export interface PADInfo {
  bankAccountNumber: string
  bankInstitutionNumber: string
  bankTransitNumber: string
  isTOSAccepted?: boolean
  isAcknowledged?: boolean
}

export interface CFSAccountDetails {
  bankAccountNumber: string
  bankInstitutionNumber: string
  bankTransitNumber: string
  cfsAccountNumber: string
  cfsPartyNumber: string
  cfsSiteNumber: string
  status: string
}

export interface GLInfo {
  client: string
  responsibilityCentre: string
  serviceLine: string
  stob: string
  projectCode: string
}

export interface OrgPaymentDetails {
  accountId: string
  accountName: string
  bcolAccount: string
  bcolUserId: string
  cfsAccount: CFSAccountDetails
  obCredit: string
  padCredit: string
  paymentMethod: string
  statementNotificationEnabled: true
  padTosAcceptedBy:string
  padTosAcceptedDate:string
  futurePaymentMethod: string
  revenueAccount?: GLInfo
  eftEnable?: boolean
}

export interface RoleInfo {
  icon: string
  name: string
  desc: string
  displayName: string
  displayOrder: number
  label: string
}

export interface AddUserBody {
  username: string
  password: string
  selectedRole?: RoleInfo
  membershipType: string
}

export interface AddUsersToOrgBody {
    users: AddUserBody[]
    orgId: string
}

export interface BulkUserResponse {
  httpStatus: number
  username: string
  created?: string
  error?: string
  firstname?: string
  type?: string
}

export interface BulkUserResponseBody {
  users: BulkUserResponse[]
}

export enum MembershipStatus {
  'Active' = 'ACTIVE',
  'Inactive' = 'INACTIVE',
  'Rejected' = 'REJECTED',
  'Pending' = 'PENDING_APPROVAL',
  'PendingStaffReview'='PENDING_STAFF_REVIEW'
}

export enum MembershipType {
  'Admin' = 'ADMIN',
  'Coordinator' = 'COORDINATOR',
  'User' = 'USER',
  'Staff' = 'STAFF',
  'ExternalStaff' = 'EXTERNAL_STAFF'
}

export interface UserTerms {
    isTermsOfUseAccepted: boolean
    termsOfUseAcceptedVersion: string
}

export interface User {
    firstname: string;
    lastname: string;
    username?: string;
    modified?: Date
    userTerms?: UserTerms
    contacts?: Contact[]
    email?: string,
    loginSource?: string
    id?: number
    keycloakGuid?: string
    verified?:boolean
}

export interface Member {
  id: number
  membershipTypeCode: MembershipType
  membershipStatus: MembershipStatus
  roleDisplayName?:string
  user: User
}
