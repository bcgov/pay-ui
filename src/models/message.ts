export interface Message {
  text: string
  type: 'success' | 'error'
  show: boolean
}
