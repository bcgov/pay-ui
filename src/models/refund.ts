export interface RefundRevenueType {
  paymentLineItemId: string
  refundAmount: number
  refundType: string
}

export interface RefundRequest {
  reason: string
  refundRevenue?: RefundRevenueType[]
}

export interface EftRefundBaseRequest {
  shortNameId: number
  refundAmount: number
  refundEmail: string
  comment?: string
}

export interface EftRefundRequest extends EftRefundBaseRequest{
  casSupplierNum: string
  casSupplierSite: string
}

export interface EftRefundChequeRequest extends EftRefundBaseRequest{
  entityName: string
  addressCity: string
  addressCountry: string
  addressRegion: string
  deliveryInstructions?: string
  postalCode: string
  streetAddress: string
  streetAddressAdditional?: string
}
