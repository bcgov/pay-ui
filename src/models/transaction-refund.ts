import { Detail } from './Invoice'

export interface PaymentData {
  accountName: string | null
  folioNumber: string | null
  initiatedBy: string | null
  paymentMethod: string | null
  paymentStatus: string | null
  totalTransactionAmount: number
}

export interface TransactionData {
  invoiceId: number | null
  transactionDate: string | null
  invoiceReferenceId: string | null
  invoiceStatusCode: string | null
  invoiceCreatedOn: string | null
  transactionAmount: number
  applicationName: string | null
  applicationType: string | null
  businessIdentifier: string | null
  applicationDetails: Detail[] | null
}

export enum RefundType {
  FULL_REFUND = 'Full Refund',
  PARTIAL_REFUND = 'Partial Refund'
}

export interface RefundLineItem {
  id: number | null
  description: string | null
  filingFees?: number | null
  serviceFees?: number | null
  priorityFees?: number | null
  total?: number | null
  futureEffectiveFees?: number | null
  filingFeesRequested?: number | string | null
  serviceFeesRequested?: number | string | null
  priorityFeesRequested?: number | string | null
  futureEffectiveFeesRequested?: number | string | null
  refundEntireItemRequested?: boolean | null
}

export interface RefundFormData {
  refundType: RefundType | null
  refundLineItems: RefundLineItem[] | null
  totalRefundAmount: number | null
  refundMethod: string | null
  notificationEmail: string | null
  reasonsForRefund: string | null
  staffComment: string | null
  requestedBy: string | null
  requestedTime: string | null
}

export enum RefundRequestStage {
  REQUEST_FORM = 'Request Form',
  DATA_VALIDATED = 'DATA Validated'
}
