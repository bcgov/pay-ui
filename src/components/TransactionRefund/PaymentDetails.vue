<template>
  <v-card flat>
    <v-card-title class="card-title">
      <v-icon
        class="pr-3"
        color="link"
      >
        mdi-file-document
      </v-icon>
      Payment Details
    </v-card-title>
    <v-card-text
      class="mt-5"
    >
      <v-row no-gutters>
        <v-col class="col-12 col-sm-12 ">
          <v-row>
            <v-col class="col-6 col-sm-3 font-weight-bold">
              Account Name
            </v-col>
            <v-col class="col-6 col-sm-9">
              {{paymentData.accountName}}
            </v-col>
          </v-row>
          <v-row>
            <v-col class="col-6 col-sm-3 font-weight-bold">
              Folio Number
            </v-col>
            <v-col class="col-6 col-sm-9">
              {{paymentData.folioNumber}}
            </v-col>
          </v-row>
          <v-row>
            <v-col class="col-6 col-sm-3 font-weight-bold">
              Initiated By
            </v-col>
            <v-col class="col-6 col-sm-9">
              {{paymentData.initiatedBy}}
            </v-col>
          </v-row>
          <v-divider class="my-6"></v-divider>
          <v-row>
            <v-col class="col-6 col-sm-3 font-weight-bold">
              Payment Info
            </v-col>
            <v-col class="col-6 col-sm-9">
              <v-row no-gutters>
                <v-col class="col-12 col-sm-6">
                  <div class="font-weight-bold">
                    Payment Method
                  </div>
                  <div class="mt-2 mb-3">
                    {{getPaymentTypeDisplayName(paymentData.paymentMethod)}}
                  </div>
                </v-col>
                <v-col class="col-12 col-sm-6">
                  <div class="font-weight-bold">
                    Status
                  </div>
                  <div class="mt-2 mb-3">
                    {{getInvoiceStatusDisplayName(paymentData.paymentStatus)}}
                  </div>
                </v-col>
              </v-row>
            </v-col>
          </v-row>
          <v-divider class="mt-2 mb-6"></v-divider>
          <v-row>
            <v-col class="col-6 col-sm-3 font-weight-bold">
              Total Amount
            </v-col>
            <v-col class="col-6 col-sm-9">
              {{formatAmount(Number(paymentData.totalTransactionAmount))}}
            </v-col>
          </v-row>
        </v-col>
      </v-row>
    </v-card-text>
  </v-card>
</template>

<script lang="ts">
import { defineComponent, PropType } from '@vue/composition-api'
import CommonUtils from '@/util/common-util'
import { PaymentData } from '@/models/transaction-refund'
import { getInvoiceStatusDisplayName } from '@/util/invoice-status-display'
import { getPaymentTypeDisplayName } from '@/util/payment-type-display'

export default defineComponent({
  name: 'PaymentDetails',
  props: {
    paymentData: {
      type: Object as PropType<PaymentData>,
      required: true
    }
  },
  setup () {
    return {
      formatAmount: CommonUtils.formatAmount,
      getInvoiceStatusDisplayName,
      getPaymentTypeDisplayName
    }
  }
})
</script>

<style lang="scss" scoped>
@import '$assets/scss/theme.scss';
@import '$assets/scss/actions.scss';

.card-title {
  background-color: $app-lt-blue;
  justify-content: left;
  height: 75px;
  font-weight: bold;
  font-size: 1.125rem;
}

</style>
