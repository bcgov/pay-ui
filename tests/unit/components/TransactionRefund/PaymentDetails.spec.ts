import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'

import PaymentDetails from '@/components/TransactionRefund/PaymentDetails.vue'
import CommonUtils from '@/util/common-util'
import { PaymentData } from '@/models/transaction-refund'

describe('PaymentDetails.vue', () => {
  const localVue = createLocalVue()
  const vuetify = new Vuetify({})

  const sampleData: PaymentData = {
    accountName: 'Acme Corp',
    folioNumber: 'FOLIO-123',
    initiatedBy: 'user@idir',
    paymentMethod: 'CHEQUE',
    paymentStatus: 'COMPLETED',
    totalTransactionAmount: 2000
  }

  beforeEach(() => {
    vi.resetModules()
    vi.clearAllMocks()
  })

  it('renders component', () => {
    const wrapper = shallowMount(PaymentDetails, {
      localVue,
      vuetify,
      propsData: { paymentData: sampleData },
      directives: { can () { } }
    })

    expect(wrapper.findComponent(PaymentDetails).exists()).toBe(true)
    expect(wrapper.text()).toContain('Payment Details')

    expect(wrapper.text()).toContain('Account Name')
    expect(wrapper.text()).toContain('Folio Number')
    expect(wrapper.text()).toContain('Initiated By')
    expect(wrapper.text()).toContain('Payment Info')
    expect(wrapper.text()).toContain('Payment Method')
    expect(wrapper.text()).toContain('Status')
    expect(wrapper.text()).toContain('Total Amount')
  })

  it('displays data values', async () => {
    const wrapper = shallowMount(PaymentDetails, {
      localVue,
      vuetify,
      propsData: { paymentData: sampleData },
      directives: { can () { } }
    })

    expect(wrapper.text()).toContain(sampleData.accountName as string)
    expect(wrapper.text()).toContain(sampleData.folioNumber as string)
    expect(wrapper.text()).toContain(sampleData.initiatedBy as string)
    expect(wrapper.text()).toContain('Cheque')
    expect(wrapper.text()).toContain('Completed')

    const expectedAmount = CommonUtils.formatAmount(sampleData.totalTransactionAmount)
    expect(wrapper.text()).toContain(expectedAmount)
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(PaymentDetails, {
      localVue,
      vuetify,
      propsData: { paymentData: sampleData },
      directives: { can () { } }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
