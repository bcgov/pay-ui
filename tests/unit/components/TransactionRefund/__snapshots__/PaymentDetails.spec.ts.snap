// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaymentDetails.vue > matches snapshot 1`] = `
"<v-card-stub data-v-6b5ebe65="" loaderheight="4" tag="div" flat="true">
  <v-card-title-stub data-v-6b5ebe65="" tag="div" class="card-title">
    <v-icon-stub data-v-6b5ebe65="" color="link" class="pr-3"> mdi-file-document </v-icon-stub> Payment Details
  </v-card-title-stub>
  <v-card-text-stub data-v-6b5ebe65="" tag="div" class="mt-5">
    <v-row-stub data-v-6b5ebe65="" tag="div" nogutters="true">
      <v-col-stub data-v-6b5ebe65="" tag="div" class="col-12 col-sm-12 ">
        <v-row-stub data-v-6b5ebe65="" tag="div">
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-3 font-weight-bold"> Account Name </v-col-stub>
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-9"> Acme Corp </v-col-stub>
        </v-row-stub>
        <v-row-stub data-v-6b5ebe65="" tag="div">
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-3 font-weight-bold"> Folio Number </v-col-stub>
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-9"> FOLIO-123 </v-col-stub>
        </v-row-stub>
        <v-row-stub data-v-6b5ebe65="" tag="div">
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-3 font-weight-bold"> Initiated By </v-col-stub>
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-9"> user@idir </v-col-stub>
        </v-row-stub>
        <v-divider-stub data-v-6b5ebe65="" class="my-6"></v-divider-stub>
        <v-row-stub data-v-6b5ebe65="" tag="div">
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-3 font-weight-bold"> Payment Info </v-col-stub>
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-9">
            <v-row-stub data-v-6b5ebe65="" tag="div" nogutters="true">
              <v-col-stub data-v-6b5ebe65="" tag="div" class="col-12 col-sm-6">
                <div data-v-6b5ebe65="" class="font-weight-bold"> Payment Method </div>
                <div data-v-6b5ebe65="" class="mt-2 mb-3"> Cheque </div>
              </v-col-stub>
              <v-col-stub data-v-6b5ebe65="" tag="div" class="col-12 col-sm-6">
                <div data-v-6b5ebe65="" class="font-weight-bold"> Status </div>
                <div data-v-6b5ebe65="" class="mt-2 mb-3"> Completed </div>
              </v-col-stub>
            </v-row-stub>
          </v-col-stub>
        </v-row-stub>
        <v-divider-stub data-v-6b5ebe65="" class="mt-2 mb-6"></v-divider-stub>
        <v-row-stub data-v-6b5ebe65="" tag="div">
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-3 font-weight-bold"> Total Amount </v-col-stub>
          <v-col-stub data-v-6b5ebe65="" tag="div" class="col-6 col-sm-9"> $2,000.00 </v-col-stub>
        </v-row-stub>
      </v-col-stub>
    </v-row-stub>
  </v-card-text-stub>
</v-card-stub>"
`;
